web3.js/src/layout.ts

```diff
@@ -135,13 +135,25 @@ export const voteInit = (property: string = 'voteInit') => {
 };
 
 export function getAlloc(type: any, fields: any): number {
-  let alloc = 0;
-  type.layout.fields.forEach((item: any) => {
+  const getItemAlloc = (item: any): number => {
     if (item.span >= 0) {
-      alloc += item.span;
+      return item.span;
     } else if (typeof item.alloc === 'function') {
-      alloc += item.alloc(fields[item.property]);
+      return item.alloc(fields[item.property]);
+    } else if ('count' in item && 'elementLayout' in item) {
+      const field = fields[item.property];
+      if (Array.isArray(field)) {
+        return field.length * getItemAlloc(item.elementLayout);
+      }
```

Comment:
This is the fix. `BufferLayout.seq` was returning an item which had `-1` for its span because the length was dynamic. It also doesn't have an `alloc` function. So instead, I added this heuristic which says that if a sequence is being encoded, check if the field being encoded is an array and use that array's length as the number of elements in the sequence. Then recurse to find the size of the element in the sequence.

Reply:
Great!